### PR TITLE
Sign in

### PR DESCRIPTION
**Pull request by YesicaNavaB**
## Summary

Sign in

## Trello Card

https://trello.com/c/vkI0ES8X

### CODE REVIEW

File: app/graphql/users/middlewares.js [hunk 1]
```diff
@@ -10,16 +11,35 @@ const createUser = async (resolve, root, args) => {
...
 
+const logIn = async (resolve, root, args) => {
+  try {
+    await schemaSignIn.validate(args.credentials, { abortEarly: false });
+  } catch (err) {
+    logger.error(err.errors);
+    throw errors.signInError(err.errors);
+  }
+  const result = await findOneUser(args.credentials.email);
+  if (result === null) {
```
Comment (gonziesc) [inline app/graphql/users/middlewares.js]: if !result

File: app/graphql/users/middlewares.js [hunk 2]
```diff
@@ -10,16 +11,35 @@ const createUser = async (resolve, root, args) => {
...
+    throw errors.signInError('user does not exist');
+  }
+  const compare = await bcrypt.compare(args.credentials.password, result.password);
+  if (compare !== true) {
```
Comment (gonziesc) [inline app/graphql/users/middlewares.js]: if (! compare) ?

File: app/graphql/users/resolvers.js [hunk 3]
```diff
@@ -19,12 +22,10 @@ const createUser = (_, { user }) =>
     });
 
 const logIn = (_, { credentials }) => {
-  // IMPORTANT: Not a functional login, its just for illustrative purposes
   userLoggedIn.publish(credentials.firstName);
   return {
-    accessToken: 'example_token',
-    refreshToken: 'example_refresh_token',
-    expiresIn: 1565990270
+    accessToken: jwt.encode({ email: credentials.email, expiresAt: Math.floor(Date.now() / 1000) }, secret),
```
Comment (gonziesc) [inline app/graphql/users/resolvers.js]: lets move this logic to another part

File: app/graphql/users/middlewares.js (hunk 1 above)
Comment (YesicaNavaB) [reply app/graphql/users/middlewares.js]: ok

File: app/graphql/users/resolvers.js (hunk 3 above)
Comment (YesicaNavaB) [reply app/graphql/users/resolvers.js]: ok!

**Review by gonziesc — APPROVED** [review]

**Pull request merged**